rust/nasl-interpreter/src/built_in_functions/ssh.rs

```diff
@@ -0,0 +1,2107 @@
+                        }
+                        Err(_) => {
+                            return Err(FunctionErrorKind::Diagnostic(
+                                format!(
+                                    "Failed setting user authentication for SessionID {}",
+                                    session_id
+                                ),
+                                Some(NaslValue::Null),
+                            ));
+                        }
+                    }
+                }
+                return Ok(NaslValue::String(prompt));
+            }
+            Ok(NaslValue::Null)
+        }
+        _ => Err(FunctionErrorKind::Diagnostic(
+            format!("Session ID {} not found", session_id),
+            Some(NaslValue::Null),
+        )),
+    }
+}
+
+/// Authenticate a user on an ssh connection
+///
+/// The function finishes the authentication process started by
+/// ssh_login_interactive. The function expects the session id as its first
+/// unnamed argument.
+///
+/// To finish the password, the named argument "password" must contain
+/// a password.
+///
+/// nasl params
+///
+/// - An SSH session id.
+///
+/// nasl named params
+///
+/// - password: A string with the password.
+///
+/// return An integer as status value; 0 indicates success.
+///
+fn nasl_ssh_login_interactive_pass<K>(
+    register: &Register,
+    ctx: &Context<K>,
+) -> Result<NaslValue, FunctionErrorKind> {
+    let positional = register.positional();
+    if positional.is_empty() {
+        return Err(FunctionErrorKind::MissingPositionalArguments {
+            expected: 0,
+            got: 1,
+        });
+    }
+
+    let session_id = match &positional[0] {
+        NaslValue::Number(x) => *x as i32,
+        _ => {
+            return Err(FunctionErrorKind::WrongArgument(
+                ("Invalid session ID").to_string(),
+            ))
+        }
+    };
+
+    let password = match register.named("pass") {
+        Some(ContextType::Value(NaslValue::String(x))) => x,
+        _ => return Err(FunctionErrorKind::from("pass")),
+    };
+
+    let mut sessions = lock_sessions(&ctx.sessions().ssh_sessions)?;
+    match sessions
+        .iter_mut()
+        .enumerate()
+        .find(|(_i, s)| s.session_id == session_id)
+    {
+        Some((_i, session)) => {
+            let verbose = session.verbose > 0;
+            let info = match session.session.userauth_keyboard_interactive_info() {
+                Ok(i) => i,
+                Err(_) => {
+                    return Err(FunctionErrorKind::Diagnostic(
+                        format!(
+                            "Failed setting user authentication for SessionID {}",
+                            session_id
+                        ),
+                        Some(NaslValue::Number(-1)),
+                    ));
+                }
+            };
+
+            if verbose {
+                ctx.logger().info(&format!("SSH kbdint name={}", info.name));
```

Comment:
because this is use an ENV variable to switch it on, different from the general debug level. It will only be available if it is set via, which also turns the libssh debug log on     
```
let verbose = env::var("OPENVAS_LIBSSH_DEBUG")